server/src/main/scala/org/http4s/server/middleware/PushSupport.scala

```diff
@@ -52,30 +49,27 @@ object PushSupport {
       if (verify(v.location)) {
         val newReq = locToRequest(v, req)
         if (v.cascade) facc.flatMap { accumulated => // Need to gather the sub resources
-          try {
-            route
-              .mapF[OptionT[F, ?], Vector[PushResponse[F]]] {
-                _.semiflatMap { response =>
-                  response.attributes
-                    .get(pushLocationKey)
-                    .map { pushed =>
-                      collectResponse(pushed, req, verify, route).map(
-                        accumulated ++ _ :+ PushResponse(v.location, response))
-                    }
-                    .getOrElse(F.pure(accumulated :+ PushResponse(v.location, response)))
-                }
+          route
+            .mapF[OptionT[F, ?], Vector[PushResponse[F]]] {
+              _.semiflatMap { response =>
+                response.attributes
+                  .get(pushLocationKey)
+                  .map { pushed =>
+                    collectResponse(pushed, req, verify, route).map(
+                      accumulated ++ _ :+ PushResponse(v.location, response))
+                  }
+                  .getOrElse(F.pure(accumulated :+ PushResponse(v.location, response)))
               }
-              .apply(newReq)
-              .getOrElse(Vector.empty[PushResponse[F]])
-          } catch { case t: Throwable => handleException(t); facc }
+            }
+            .apply(newReq)
+            .getOrElse(Vector.empty[PushResponse[F]])
```

Comment:
I'm trying to think what exception here we were motivated to catch in the first place.

Reply:
I Don't think this one is doing anything in particular.. its removal didn't break anything.

Reply:
Kill it.